Yoast/wpseo-woocommerce
# PHPCS: Update to YoastCS 2.0.0

## Summary
This PR can be summarized in the following changelog entry:

* The `wpseo_woocommerce_og_price` filter hook has been deprecated in favour of the `Yoast\WP\Woocommerce\og_price` hook.


## Relevant technical choices:

### PHPCS: Update to YoastCS 2.0.0

This updates the dependency in `composer.json`, as well as the PHPCS ruleset for YoastCS 2.0.0.

Includes:
* A selective update of the `composer.lock` for just YoastCS and its dependencies.
* Removing the tweaks to the Composer `check-cs` script to allow for different PHP requirements in different directories.
    PHP 5.6 is now the minimum PHP version everywhere.
* Removing running the `config-yoastcs` script from the Travis script.
    This hasn't been needed since YoastCS 1.1.0.

Ref: https://github.com/Yoast/yoastcs/releases/tag/2.0.0

### PHPCS: remove whitelist comment

... which is no longer needed now PHP 5.6 is the minimum PHP version.

### PHPCS: whitelist use of YoastSEO hooks

### CS: deprecated old-style hook and add new-style hook

This deprecates the `wpseo_woocommerce_og_price` filter hook in favour of the `Yoast\WP\Woocommerce\og_price` hook.

[This needs a changelog entry]

### CS: use short arrays (everywhere)

### CS: don't use Yoda conditions 

## Test instructions

This PR can be tested by following these steps:

### Test the updated composer script:
* Check out this branch
* Run `composer install`
* Run `composer check-cs` and see a clean result.

### Test the hook deprecation
* Start on `trunk` & make sure `WP_DEBUG`/PHP error logging is turned on.
* Add a minimal hook-in for the `wpseo_woocommerce_og_price` filter either in a custom plugin or the theme's `functions.php` file.
    Something along the lines of the below should do:
    ```php
    function test_deprecated_filter($input){
        return $input;
    }
    add_filter('wpseo_woocommerce_og_price', 'test_deprecated_filter');
    ```
* Open a Woocommerce product page with images. There should be no errors in the logs.
* Switch to this branch.
* Refresh the page. A deprecation notice along the lines of the following should show up in the error logs: _'wpseo_woocommerce_og_price is <strong>deprecated</strong> since version Yoast WooCommerce 12.5.0! Use Yoast\WP\Woocommerce\og_price instead.'_
* Now change the `add_filter()` call as added above to:
    ```php
    add_filter('Yoast\WP\Woocommerce\og_price', 'test_deprecated_filter');
    ```
* Refresh the page and the error should longer show up.
* Look at the OpenGraph tags of the page, particularly the `product:price:amount` and `product:price:currency` tags and take note of the price being included in the tags.
* Now, change the `return $input;` in the filter to `return false`.
* Refresh the page again and check that the above mentioned tags are no longer included in the OpenGraph tags anymore.
